### DISCUSSION TITLE
Overall format/structure - incorporating best practices from UPI API/Tech Spec doc

### DISCUSSION THREAD

**Discussion by rahul101001000** [Sunbird-RC/community#65] [Sunbird RC Technical Specification Feedback]
The UPI API/Tech spec doc ([link](https://www.mygov.in/digidhan/pages/pdf/sbi/NPCI%20Unified%20Payment%20Interface.pdf)) is one of the easiest to understand. See section 2.6 that gives use cases (external and internal view), followed by sequence diagrams and failure scenarios in section 3.1, and the API examples in 3.4. This structure ensures the information provided is comprehensive, logically derived from the previous section, and most importantly anchored in use cases.

Could we use elements of this structure/format for the technical specification for Sunbird-RC. Right now the spec misses out on readability and comprehensiveness. Happy to help in any way. @pramodkvarma

**Comment by pramodkvarma** [comment]
Sure, do you want to contribute this? You can edit the specs page (your version) and send for review. We can then do a PR to bring it in.

**Comment by rahul101001000** [comment]
Sure @pramodkvarma, let me start with editing one page - [Edu use cases](https://docs.sunbirdrc.dev/example-use-cases/edu-registries) and express use cases in the following format - 
1. End user experience flow
2. Behind the scenes flow
3. Sequence diagram with API calls and schema references
4. Pointer to example code

If this is useful then can structure other pages better as well.

**Reply by pramodkvarma** [reply]
Please do check out existing work by @jayadinni and @kshitij-design and build on top of that

**Comment by rahul101001000** [comment]
Most definitely. Please do send any pointers. Right now seeing only a
skeleton doc on the link I shared.

**Reply by jayadinni** [reply]
@rahul101001000 Here is [high level user flow](https://docs.sunbirdrc.dev/example-use-cases/edu-registries/user-flow)

**Comment by rahul101001000** [comment]
This is great, thank you @jayadinni ! 

I was looking for something like the [DEPA sequence diagrams](https://github.com/iSPIRT/DEPA). What I like about this is that the arrows have not only the API call being made but also the schema for the object being passed. 

Maybe in the current diagram itself, API endpoint and schema may be added?


Also I'm not finding the API docs or schemas, could you please point to them. 

Thanks in advance, 

Regards
Rahul
